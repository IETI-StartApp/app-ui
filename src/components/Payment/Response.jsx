import React, { useEffect } from 'react';
import { useStyles } from './styles';
import { Table, TableBody, TableCell, TableContainer, TableRow, Grid, Box, Divider, Typography, Paper } from '@material-ui/core';

export default function Response() {

    const classes = useStyles();

    useEffect(() => {
        let query = new URLSearchParams(window.location.search);
        fetch(`https://secure.epayco.co/validation/v1/reference/${query.get("ref_payco")}`)
            .then(response => {
                if (!response.ok) throw new Error('Response is NOT ok');
                return response.json();
            })
            .catch(() => console.log("error"))
            .then(response => {
                document.getElementById("ref").innerText = response.data.x_id_invoice;
                document.getElementById("date").innerText = response.data.x_transaction_date;
                document.getElementById("reason").innerText = response.data.x_response_reason_text;
                document.getElementById("bank").innerText = response.data.x_bank_name;
                document.getElementById("transaction").innerText = response.data.x_transaction_id;
                document.getElementById("total").innerText = response.data.x_amount + ' ' + response.data.x_currency_code;
            });
    }, []);

    return (
        <Grid item sm={8}>
            <Box mt={2} mb={4}>
                <Grid container justify="center">
                    <img className="center-block" src="https://369969691f476073508a-60bf0867add971908d4f26a64519c2aa.ssl.cf5.rackcdn.com/btns/epayco/logo1.png" alt="logo"/>
                </Grid>
            </Box>
            <Box mb={3}>
                <Grid>
                    <Box p={2} mb={2}>
                        <Typography  variant="h6" component="h6">Respuesta de la Transacci√≥n</Typography>
                        <Divider></Divider>
                    </Box>
                    <TableContainer component={Paper}>
                        <Table className={classes.table} size="small" aria-label="a dense table">
                            <TableBody>
                                <TableRow>
                                    <TableCell component="th" scope="row">Referencia</TableCell>
                                    <TableCell align="right" id="ref"></TableCell>
                                </TableRow>
                                <TableRow>
                                    <TableCell component="th" scope="row">Fecha</TableCell>
                                    <TableCell align="right" id="date"></TableCell>
                                </TableRow>
                                <TableRow>
                                    <TableCell component="th" scope="row">Motivo</TableCell>
                                    <TableCell align="right" id="reason"></TableCell>
                                </TableRow>
                                <TableRow>
                                    <TableCell component="th" scope="row">Banco</TableCell>
                                    <TableCell align="right" id="bank"></TableCell>
                                </TableRow>
                                <TableRow>
                                    <TableCell component="th" scope="row">Recibo</TableCell>
                                    <TableCell align="right" id="transaction"></TableCell>
                                </TableRow>
                                <TableRow>
                                    <TableCell component="th" scope="row">Total</TableCell>
                                    <TableCell align="right" id="total"></TableCell>
                                </TableRow>
                            </TableBody>
                        </Table>
                    </TableContainer>
                </Grid>
            </Box>
            <Grid container justify="center">
                <img src="https://369969691f476073508a-60bf0867add971908d4f26a64519c2aa.ssl.cf5.rackcdn.com/btns/epayco/pagos_procesados_por_epayco_260px.png" alt="epaycoimg"/>
                <img src="https://369969691f476073508a-60bf0867add971908d4f26a64519c2aa.ssl.cf5.rackcdn.com/btns/epayco/credibancologo.png" height="40px" alt="epaycoimg2"/>
            </Grid>
    </Grid>
    );
}