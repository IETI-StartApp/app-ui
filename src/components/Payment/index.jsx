import React, { useEffect } from 'react';

export default function PayButton({amount, projectname, currency, country}) {

    useEffect(() => {
        const s = document.createElement('script');
        s.src = "https://checkout.epayco.co/checkout.js";
        s.classList.add("epayco-button");
        s.setAttribute('data-epayco-key', "705b5bed02e3e22f550dfc97fdb2de2a");
        s.setAttribute('data-epayco-private-key', "76fe4aaa1859a5e172a4ccecca6d6d27");
        s.setAttribute('data-epayco-amount', amount);
        s.setAttribute('data-epayco-name', "Inversi√≥n");
        s.setAttribute('data-epayco-description', projectname);
        s.setAttribute('data-epayco-currency', currency);
        s.setAttribute('data-epayco-country', country);
        s.setAttribute('data-epayco-test', "true");
        s.setAttribute('data-epayco-external', "false");
        s.setAttribute('data-epayco-response', "http://localhost:3000/epayco/response");
        s.setAttribute('data-epayco-confirmation', "http://localhost:3000/epayco/confirmation");
        document.getElementById("PayButtonDiv").append(s);
    }, [amount, projectname, currency, country]);

    return (
        <div id="PayButtonDiv" />
    );
}
