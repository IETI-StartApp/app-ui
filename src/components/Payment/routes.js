const express = require('express');
const crypto = require('crypto');
const router = express.Router();

router.get("/api/something", (req, res) => {
  res.json({ status: "OK" })
})


router.post("/epayco/confirmation", (req, res) => {

  const p_cust_id_cliente = '513409';
  const p_key = '7b7246effa7d528a496888901a4d43f5b0305309';
  const x_ref_payco = req.query['x_ref_payco'];
  const x_transaction_id = req.query['x_transaction_id'];
  const x_amount = req.query['x_amount'];
  const x_currency_code = req.query['x_currency_code'];
  const x_signature = req.query['x_signature'];

  const str = `${p_cust_id_cliente}^${p_key}^${x_ref_payco}^${x_transaction_id}^${x_amount}^${x_currency_code}`;
  var hash = crypto.createHash('sha256').update(str).digest('hex');
  if (hash != x_signature) {
    res.sendStatus(422);
    return;
  }
  const x_cod_response = req.query['x_cod_response'];
  switch (x_cod_response) {
    case 1:
      console.log("transacci贸n aceptada");
      break;
    case 2:
      console.log("transacci贸n rechazada");
      break;
    case 3:
      console.log("transacci贸n pendiente");
      break;
    case 4:
      console.log("transacci贸n fallida");
      break;
  }
  res.send("OK");
})

module.exports = router;