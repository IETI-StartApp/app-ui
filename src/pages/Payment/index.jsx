import React from 'react';
import Response from '../../components/Payment/Response';
import Navigation from '../../components/Navbar/Navigation';
import { useStyles } from './styles';

export const PaymentResponse = () => {
    const classes = useStyles();
    return (
        <>
            <Navigation />
            <br />
            <div className={classes.root}>
                <Response />
            </div>
        </>
    );
};