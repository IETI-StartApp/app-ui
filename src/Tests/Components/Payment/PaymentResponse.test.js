import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import React from 'react';
import { PaymentResponse } from '../../../pages/Payment/index';

describe('Testing PaymentResponse', () => {

    const wrapper = shallow(<PaymentResponse />)

    test('should load PaymentResponse', () => {
        expect(wrapper).toMatchSnapshot();
    });

})