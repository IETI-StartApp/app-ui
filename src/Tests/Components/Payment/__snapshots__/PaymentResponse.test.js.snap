// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing PaymentResponse should load PaymentResponse 1`] = `
<Fragment>
  <Navigation />
  <br />
  <div
    className="makeStyles-root-1"
  >
    <Response />
  </div>
</Fragment>
`;
